In [11]:
import pandas as pd
import numpy as np

In [4]:
test_prep=pd.read_csv('dataset w7.csv')

# 1. Carry out descriptive analytics on the dataset.

In [5]:
test_prep.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

# 2. Check whether female and male students scored the same marks.

In [15]:
import scipy.stats as stats
stats.ttest_ind(test_prep['Total Marks'][test_prep['Gender'] == 'male'],
                 test_prep['Total Marks'][test_prep['Gender'] == 'female'])

Ttest_indResult(statistic=-0.6940885268848668, pvalue=0.4937822317960763)

    Interpretation
    
    Since we have to prove the marks are not same,
    our Null Hypothesis is 'The marks scored by male and female students are not same'.
    
    degrees of freedom = n-2 = 28-2 = 26
    significance level = 5% = 0.05
    corresponding t-score = 2.056
    
    Here, |t-score|=0.69<2.056
    Therefore, we cannot reject the null hypothesis.
    
    Also, p-value=0.49>0.05
    Therefore, we cannot reject the null hypothesis.
    
    That is, the marks are not same.


# 3. Whether test preparation helps the students?

In [14]:
import scipy.stats as stats
stats.ttest_ind(test_prep['Total Marks'][test_prep['Test preparation'] == 'none'],
                 test_prep['Total Marks'][test_prep['Test preparation'] == 'completed'])

Ttest_indResult(statistic=-3.6143637527769217, pvalue=0.0012671292457320106)

    Interpretation

    Since we have to prove the test preparation helps the students,
    our Null Hypothesis is 'Test preparation does not help the students'.

    degrees of freedom = n-2 = 28-2 = 26
    significance level = 5% = 0.05
    corresponding t-score = 2.056

    Here, |t-score|=3.614>2.056
    Therefore, we can reject the null hypothesis.

    Also, p-value=0.001<0.05
    Therefore, we can reject the null hypothesis.

    That is, Test preparation actually helps the students.